falcon/asgi_spec.py

```diff
@@ -13,6 +13,7 @@
 # limitations under the License.
 
 """Constants, etc. defined by the ASGI specification."""
+from __future__ import annotations
```

Comment:
in general `from __future__ import annotations` makes sense to have since
- it's how python will eventually work
- substantially speeds up the import of the module, since no type needs to be resolved, all annotations are just strings
- avoids the need of manually using string every time a forward-ref or type only import is referenced

overall, in my opinion, it makes sense to have in every module that uses type checking in any significant capacity

Reply:
Agreed, it reduces the import complexity and has no impact on runtime performance, so I only see the positives as well.

Reply:
there is also a flake8 plugin the enforces this. maybe we could look into it at some point?

it's called `flake8-future-annotations`